src/Analyzer.Cli/CommandLineParser.cs

```diff
@@ -80,7 +94,7 @@ private RootCommand SetupCommandLineAPI()
             return rootCommand;
         }
 
-        private int AnalyzeTemplate(FileInfo templateFilePath, FileInfo parametersFilePath, bool printMessageIfNotTemplate = true)
+        private int AnalyzeTemplate(FileInfo templateFilePath, FileInfo parametersFilePath, ReportFormat reportFormat, FileInfo outputFilePath, bool printMessageIfNotTemplate = true, IReportWriter writer = null)
```

Comment:
the tool can analyze a directory or a template file.
the existing function for analyze directory is to enumerate all template files under the directory then call AnalyzeTemplate for each template file.

If tool analyzes a directory, the writer is initialized at AnalyzeDirectory function, and pass the write to AnalyzeTemplate function so it will not initialize a new writer again.

If tool analyzes a template file, it call AnalyzeTemplate with null writer, so the function will create its own writer in this case.